File: dashboard/src/stores/databases.ts
import { atom } from 'nanostores';

export type ManagedDb = {
  id: string;
  name: string;
  db_type: string;
  container_id: string | null;
  credentials: string;
  backup_schedule: string | null;
  app_id: string | null;
  created_at: string;
}

export const $databases = atom<ManagedDb[]>([]);
export const $databasesLoaded = atom(false);

File: src/db/models/databases.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ManagedDatabase {
    pub id: String,
    pub name: String,
    pub db_type: String,
    pub container_id: Option<String>,
    pub credentials: String,
    pub backup_schedule: Option<String>,
    pub app_id: Option<String>,
    pub project_id: Option<String>,
    pub backup_retention_count: i32,
    pub created_at: String,
}

pub struct NewManagedDatabase {
    pub name: String,
    pub db_type: String,
    pub app_id: Option<String>,
}
